src/openrct2-ui/windows/Research.cpp

```diff
@@ -154,117 +113,218 @@ static constexpr const StringId ResearchStageNames[] = {
     STR_RESEARCH_STAGE_UNKNOWN,
 };
 
-static void WindowResearchSetPage(WindowBase* w, int32_t page);
-static void WindowResearchSetPressedTab(WindowBase* w);
-static void WindowResearchDrawTabImages(DrawPixelInfo& dpi, WindowBase* w);
-
-WindowBase* WindowResearchOpen()
+class ResearchWindow final : public Window
 {
-    WindowBase* w;
-
-    w = WindowBringToFrontByClass(WindowClass::Research);
-    if (w == nullptr)
+    void OnOpen() override
     {
-        w = WindowCreateAutoPos(WW_FUNDING, WH_FUNDING, window_research_page_events[0], WindowClass::Research, WF_10);
-        w->widgets = window_research_page_widgets[0];
-        w->number = 0;
-        w->page = 0;
-        w->frame_no = 0;
-        w->disabled_widgets = 0;
+        SetPage(WINDOW_RESEARCH_PAGE_DEVELOPMENT);
```

Comment:
I think this is not working as intended. `OnOpen` is just called when the window is not created, the previous code would reset the page to 0 if the window was already open, but on funding. I think if you test your build and click on the research button, it will bring it to focus, but keep on the funding tab. Can you test?

If my assumption is correct, then you won't be able to use `WindowFocusOrCreate`

Reply:
Yes, you are correct. It doesn't change to the development tab if it is already open. I have also confirmed that it does switch to the development tab in the current release.

I can't find any other windows with this behaviour to use as an example. Are there any alternatives to `WindowFocusOrCreate`? 

I had a look at the methods in WindowBase and couldn't see any to override that looked appropriate.

Would it be reasonable to manually call `SetPage` after `WindowFocusOrCreate`?

Reply:
Ok, I'll go with `SetPage` then.